### PR TITLE
JMS changing polling logic

### PR DESCRIPTION
**Pull request by eyalkoren**
Implementing first bullet of elastic/apm-agent-java#664 

- [x] Implement code
- [x] Add tests
- [ ] ~~Update documentation~~ will be done with #876 
- [x] Update [CHANGELOG.md](CHANGELOG.md)

### CODE REVIEW

**Comment by eyalkoren** [comment]
With the new logic, Spring JMS test passes when the MessageListener instrumentation is disabled, validating that this logic is viable at least in this case (Spring actually uses polling APIs under the hood and applies logic to invoke a JMS-like `onMessage` API).

**Comment by eyalkoren** [comment]
Jenkins test this please

**Comment by eyalkoren** [comment]
Following a discussion, required changes:
- Use only a single type of transaction: `messaging`.
- Don't create polling transactions by default, but only handling ones.
- Try to find a way to reduce risk of leaking unclosed transactions, for example- instrument (or wrap) all Runnables and Callables so that at the end of `run`/`call` - we end any active transaction. This must not interfere with the Executors' plugin and beware of added related overhead.

File: apm-agent-plugins/apm-jms-plugin/src/test/java/co/elastic/apm/agent/jms/JmsInstrumentationIT.java [hunk 1]
```diff
@@ -112,87 +123,211 @@ public void endTransaction() throws Exception {
...
-        Message incomingMessage = receiveOnNonTracedThread(receiveMethod).get(3, TimeUnit.SECONDS);
+        Message incomingMessage = resultQ.poll(2, TimeUnit.SECONDS);
+        assertThat(incomingMessage).isNotNull();
         verifyMessage(message, incomingMessage);
-        verifySendReceiveOnNonTracedThread(queue.getQueueName(), 1);
+        verifySendReceiveOnNonTracedThread(queue.getQueueName());
     }
 
-    private void verifySendReceiveOnNonTracedThread(String destinationName, int expectedReadTransactions) {
+    private void verifySendListenOnNonTracedThread(String destinationName, int expectedReadTransactions) {
```
Comment (felixbarny) [inline apm-agent-plugins/apm-jms-plugin/src/test/java/co/elastic/apm/agent/jms/JmsInstrumentationIT.java]: Could we use [awaitility](https://github.com/awaitility/awaitility) in order to get rid of some `Thread.sleep`s? 

```java
await().atMost(500, MILLISECONDS).until(() -> !reporter.getSpans().isEmpty());
```

Sleeps are not only a source of flakiness but also make it hard to follow on which condition we are waiting for.
